src/Twig/Extension/TemplateRegistryExtension.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Twig\Extension;
+
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Admin\Pool;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+final class TemplateRegistryExtension extends AbstractExtension
+{
+    /**
+     * @var Pool
+     */
+    private $pool;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    public function __construct(Pool $pool, RequestStack $requestStack)
+    {
+        $this->pool = $pool;
+        $this->requestStack = $requestStack;
+    }
+
+    public function getFunctions()
+    {
+        return [
+            new TwigFunction('get_admin_template', [$this, 'getTemplate']),
+            new TwigFunction('get_admin_pool_template', [$this, 'getPoolTemplate']),
+        ];
+    }
+
+    /**
+     * @param string $name
+     *
+     * @return null|string
+     */
+    public function getTemplate($name)
+    {
+        return $this->getAdmin()->getTemplate($name);
+    }
+
+    /**
+     * @param string $name
+     *
+     * @return null|string
+     */
+    public function getPoolTemplate($name)
+    {
+        return $this->pool->getTemplate($name);
+    }
+
+    /**
+     * @return AdminInterface|false|null
+     */
+    private function getAdmin()
+    {
+        return $this->pool->getAdminByAdminCode(
+            $this->requestStack->getCurrentRequest()->get('_sonata_admin')
```

Comment:
Maybe this should get saved somehow?

Reply:
Meh

Reply:
I mean the whole getAdmin, could it be called many times on a single request?

Reply:
It could, but is it that slow?

Reply:
Maybe `$this->pool->getInstance()` would be better?

Reply:
It depends if the admin code can contain pipes or not, and I think it could in the case of children admins?

Reply:
In the case of `parent|child`, what admin should provide the template? `child`, right?

Reply:
Yes, you're right. Child should provide the template in that case.

Reply:
Caching this in a private attribute will probably be best, then. It could save a lot of calls to the pool. And the resulting admin is always the same anyway.
Correct me if I'm wrong, but the `_sonata_admin` parameter in the request should never change during a single request, right?

Reply:
Please try your PR on a real project and tell us if you manage to measure any difference. If you don't I'd rather not make any assumptions.

Reply:
Will do.

Reply:
From what I could see, it saves very little to no time to cache the admin locally. I think it's best to keep this like it is now, at least for the time being.
The admin itself is pulled from the pool, so it only returns a reference.

Reply:
That's what I though, it's not that expensive.